guide/client_service/client_main.bal

```diff
@@ -0,0 +1,235 @@
+        io:println("3. Delete a student");
+        io:println("4. Make a mock error");
+        io:println("5: Get a student's marks");
+        io:println("6. Exit \n");
+
+        // Read user's choice.
+        string choice = io:readln("Enter choice 1 - 5: ");
+        if (!isInteger(choice)){
+            io:println("Choice must be of a number");
+            io:println();
+            continue;
+        }
+
+        operation = check <int>choice;
+        // Program runs until the user inputs 6 to terminate the process.
+        if (operation == 6) {
+            break;
+        }
+        // User chooses to add a student.
+        if (operation == 1) {
+            addStudent(req);
+        }
+        // User chooses to list down all the students.
+        else if (operation == 2) {
+            viewAllStudents();
+        }
+        // User chooses to delete a student by Id.
+        else if (operation == 3) {
+            deleteStudent();
+        }
+        // User chooses to make a mock error.
+        else if (operation == 4) {
+            makeError();
+        }
+        else if (operation == 5){
+          getMarks();
+        }
+        else {
+            io:println("Invalid choice");
+        }
+    }
+}
+
+function isInteger(string input) returns boolean {
+    string regEx = "\\d+";
+    boolean isInt = check input.matches(regEx);
+    return isInt;
+}
+
+function addStudent(http:Request req) {
+    // Get student name, age mobile number, address.
+    var name = io:readln("Enter Student name: ");
+    var age = io:readln("Enter Student age: ");
+    var mobile = io:readln("Enter mobile number: ");
+    var add = io:readln("Enter Student address: ");
+
+    // Create the request as json message.
+    json jsonMsg = { "name": name, "age": check <int>age, "mobNo": check <int>mobile, "address": add };
+    req.setJsonPayload(jsonMsg);
+
+    // Send the request to students service and get the response from it.
+    var resp = studentData->post("/records/addStudent", req);
+    match resp {
+        http:Response response => {
+            var msg = response.getJsonPayload();
+            //obtaining the result from the response received
+            match msg {
+                json jsonPL => {
+                    string message = "Status: " + jsonPL["Status"] .toString() + " Added Student Id :- " +
+                        jsonPL["id"].toString();
+                    // Extracting data from json received and displaying.
+                    io:println(message);
+                }
+
+                error err => {
+                    log:printError(err.message, err = err);
+                }
+            }
+        }
+        error err => {
+            log:printError(err.message, err = err);
+        }
+    }
+}
+
+function viewAllStudents() {
+    // Sending a request to list down all students and get the response from it.
+    var requ = studentData->post("/records/viewAll", null);
```

Comment:
Can we use a better name instead of `requ`?

Reply:
Please fix in other places too.

Reply:
Yeah Noted. I will make the changes.